flink-connectors/flink-connector-base/src/test/java/org/apache/flink/connector/base/source/reader/SourceReaderBaseTest.java

```diff
@@ -286,4 +291,71 @@ protected TestingSourceSplit toSplitType(String splitId, TestingSourceSplit spli
 
 		return reader;
 	}
+
+	@Test
+	public void pollNextReturnMoreAvailableWhenAllSplitFetcherCloseWithLeftoverElementInQueue()
```

Comment:
Usually the test method names starts with `testXXX`. I'll fix this upon checking in the patch.